src/Opserver.Poller/Services/PollSql.cs

```diff
@@ -0,0 +1,157 @@
+        private readonly IElasticClient _elasticClient;
+        private readonly ILogger<IPollSql> _logger;
+
+        public PollSql(SQLModule sqlModule, IElasticClient elasticClient, ILogger<IPollSql> logger)
+        {
+            _sqlModule = sqlModule;
+            _elasticClient = elasticClient;
+            _logger = logger;
+        }
+
+        public void ObserveAllInstances()
+        {
+            var sqlInstances = _sqlModule.AllInstances;
+
+            foreach (var sqlInstance in sqlInstances)
+            {
+                _logger.LogInformation("Observing " + sqlInstance.Name);
+                sqlInstance.Polled += SqlInstanceOnMonitorStatusChanged;
+            }
+        }
+
+        private void SqlInstanceOnMonitorStatusChanged(object sender, PollNode.PollResultArgs a)
+        {
+            _logger.LogDebug("Polled");
+            if (!(sender is SQLInstance sqlInstance))
+            {
+                return;
+            }
+
+            if (sqlInstance.ServerProperties.Data == null)
+            {
+                _logger.LogWarning($"No ServerProperties data for ${sqlInstance.Name} skipping ");
+            }
+
+            var esDoc = new SqlMetricBeat()
+            {
+                InstanceName = sqlInstance.Name,
+                MachineName = sqlInstance.ServerProperties.Data.MachineName,
+                Uptime = (DateTime.UtcNow - sqlInstance.ServerProperties.Data.SQLServerStartTime).TotalSeconds,
+                Version = sqlInstance.ServerProperties.Data.Version,
+                Stats = GetStats(sqlInstance),
+                PerformancePerSec = GetPerformance(sqlInstance),
+                Memory = GetMemory(sqlInstance),
+                Cache = GetCache(sqlInstance),
+                WaitsPerSec = GetWaits(sqlInstance),
+            };
+
+            var indexResponse = _elasticClient.IndexDocument(esDoc);
+
+            if (indexResponse == null)
+            {
+                _logger.LogWarning("No response from ES");
+                return;
+            }
+
+            _logger.LogDebug($"ES: Result:{indexResponse.Result}, seqNo:{indexResponse.SequenceNumber}");
+        }
+
+        private static Dictionary<string, object> GetStats(SQLInstance i)
+        {
+            return new Dictionary<string, object>
+            {
+                ["cpu_percent"] = i.CurrentCPUPercent,
+                ["worker_count"] = i.ServerProperties.Data.CurrentWorkerCount,
+                ["session_count"] = i.ServerProperties.Data.SessionCount,
+                ["connection_count"] = i.ServerProperties.Data.ConnectionCount,
+                ["jobs_count"] = i.JobSummary.Data.Count(j => j.IsRunning)
+            };
+        }
+
+        private static Dictionary<string, object> GetWaits(SQLInstance i)
+        {
+            return new Dictionary<string, object>
+            {
+                ["lock"] = i.GetPerfCounter("Wait Statistics", "Lock waits", "Waits started per second")?.CurrentValue,
```

Comment:
Maybe these field names should be documented somehow? Ideally we could change Instance.cshtml template so that it displays this name or even provides a link to kibana dashboard. 
But I don't know how to extend this template in a way that does not break open/close principle making it a horror to maintain

Reply:
Good idea, let's look at adding this in, in the next iteration. Would be good to link it all up.